docs/react/3.optimization/error-boundaries.md

```diff
@@ -3,24 +3,27 @@ title: Error Boundaries
 sidebar_label: Error Boundaries
 ---
 
-import Img from '../../../src/components/Img';
+import Img from '../../../src/components/Img';import Hint from '../../../src/components/Hint';
 
-In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to emit cryptic errors on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them.
+In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to emit cryptic errors on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them. Like this in your console: <Img w="700" src='https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/UMQoLL.png'/>
+
+Typical symptoms include stack traces that thread through dark regions of the React internals.
 
 ## Introducing Error Boundaries
 
 A JavaScript error in a part of the UI shouldn’t break the whole app. To solve this problem for React users, React 16 introduces a new concept of an “error boundary”.
 
-Error boundaries are React components that **catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI** instead of the component tree that crashed. Error boundaries catch errors during rendering, in lifecycle methods, and in constructors of the whole tree below them.
+Error boundaries are React components that **catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI** instead of the component tree that crashed. Error boundaries catch errors during rendering, **in lifecycle methods, and in constructors of the whole tree below them**.
+
+<Img w="400" src='https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/1jJMQE.png'/>
 
-> **Note**
->
-> Error boundaries do **not** catch errors for:
->
-> - Event handlers ([learn more](https://reactjs.org/docs/error-boundaries.html#how-about-event-handlers))
-> - Asynchronous code (e.g. `setTimeout` or `requestAnimationFrame` callbacks)
-> - Server side rendering
-> - Errors thrown in the error boundary itself (rather than its children)
+<Hint type="tip">
+Error boundaries do **not** catch errors for:
```

Comment:
html tags in mdx may do not support empty line, we have to use `> Note` as for now.